### ISSUE TITLE
S3433 FP on xUnit `async void` methods

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#4861]
### Description

https://sonarsource.github.io/rspec/#/rspec/S3433

The rule raises issues on MSTest, NUnit and xUnit test methods which are `async void`. For MSTest and NUnit it is correct - they don't work (NUnit will raise an error, MSTest won't run the method at all).

However, xUnit `async void` methods run as expected and are supported by the framework (tested with xUnit 2.4.1).

### Repro steps

Checkout https://github.com/andrei-epure-sonarsource/dotnet-test-playground commit fde6fc1dfab5f2ddcf93dc0e3a1508f5cb5f03a6.

Open `UnitTest1` in VS with SonarLint (tested with SLVS 4.37).

```
[Fact]
public async void xUnit_Async_Void() =>
    await new TestAsync().Foo(o => Assert.Equal(32, o));
```

### Expected behavior

`xUnit_Async_Void` should not raise an issue.

### Actual behavior

`Make this method non-async or return "Task"` is raised.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins version - 8.27
* Visual Studio version - 16.10.3
